src/cluster_legacy.c

```diff
@@ -173,10 +173,41 @@ static inline char *clusterLinkGetNodeName(clusterLink *link) {
     return link->node ? link->node->name : "<unknown>";
 }
 
+/* By default, a server doesn't have a human-readable nodename unless explicitly
+ * assigned by CONFIG SET cluster-announce-human-nodename command or config file
+ * edit, so we simply fall back to using the node's IP and port as the nodename.
+ * This is only used for logging purpose, so we return either the SDS field or a
+ * pointer to a thread-local scratch buffer. */
+char *humanNodename(clusterNode *node) {
+    if (sdslen(node->human_nodename) > 0) {
+        return node->human_nodename;
+    }
+
+    /* Avoid allocating heap memory so that users can call the function with ease.
+     * Use a small ring of thread-local buffers here so that multiple function calls
+     * in the same logging statement are safe. */
+    enum { BUF_COUNT = 8 };
+    enum { BUF_SIZE = 64 }; /* Long enough to hold "255.255.255.255:65535" */
+    static _Thread_local char buffers[BUF_COUNT][BUF_SIZE];
+    static _Thread_local int idx;
+
+    char *buffer = buffers[idx];
+    idx = (idx + 1) % BUF_COUNT;
+
+    const int port = server.tls_cluster ? node->tls_port : node->tcp_port;
+    /* Be defensive in case IP is empty. */
+    if (node->ip[0] == '\0') {
+        snprintf(buffer, BUF_SIZE, ":%d", port);
+    } else {
+        snprintf(buffer, BUF_SIZE, "%s:%d", node->ip, port);
+    }
```

Comment:
The string length needs to be able to hold an IPv6 address, which can be up to 46 bytes. If it's an IPv6, we should wrap it in square brackets, like `[1:2:3:4:5:6:7:8]:12345`. There is a macro used for this length: `CONN_ADDR_STR_LEN`.

There is existing an existing function `formatAddr` that we can use, defined in `connection.h`.

```suggestion
    const int port = server.tls_cluster ? node->tls_port : node->tcp_port;
    formatAddr(buffer, sizeof(buffer), node->ip, port);
```

Reply:
Good call! Updated the code. Added a test case for IPv6.